src/igvfd/types/file_set.py

```diff
@@ -278,6 +287,32 @@ def assay_titles(self, request, input_file_sets=None):
     def donors(self, request, samples=None):
         return get_donors_from_samples(request, samples)
 
+    @calculated_property(
+        schema={
+            'title': 'Protocols',
+            'description': 'Links to the protocol(s) for conducting the assay on Protocols.io.',
+            'type': 'array',
+            'minItems': 1,
+            'uniqueItems': True,
+            'items': {
+                'title': 'Protocol',
+                'type': 'string',
+                'pattern': '^https://www\\.protocols\\.io/(\\S+)$'
+            },
+            'notSubmittable': True
+        }
+    )
+    def protocols(self, request, input_file_sets=None):
+        '''Calculate an array of unique protocols for all measurement sets associated with an analysis set.'''
+        protocols = set()
+        file_set_objs = get_fileset_objs_from_input_file_sets(request=request, input_file_sets=input_file_sets)
+        for file_set_obj in file_set_objs:
+            if 'MeasurementSet' in file_set_obj.get('@type'):
+                protocol = file_set_obj.get('protocols')
```

Comment:
Small change here: since protocols is not required, it is safer to add a default return value like, `file_set_obj.get('protocols', [])`